modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaGroup.es.js

```diff
@@ -0,0 +1,368 @@
+import {PropTypes} from 'prop-types';
+import CriteriaRow from './CriteriaRow.es';
+import ClayButton from '../shared/ClayButton.es';
+import ClayIcon from '../shared/ClayIcon.es';
+import {CONJUNCTIONS} from '../../utils/constants.es';
+import {DragSource as dragSource} from 'react-dnd';
+import {DragTypes} from '../../utils/drag-types.es';
+import DropZone from './DropZone.es';
+import EmptyDropZone from './EmptyDropZone.es';
+import getCN from 'classnames';
+import {
+	generateGroupId,
+	getChildGroupIds,
+	insertAtIndex,
+	replaceAtIndex
+} from '../../utils/utils.es';
+
+/**
+ * Passes the required values to the drop target.
+ * This method must be called `beginDrag`.
+ * @param {Object} props Component's current props
+ * @returns {Object} The props to be passed to the drop target.
+ */
+function beginDrag({criteria, index, parentGroupId}) {
+	const childGroupIds = getChildGroupIds(criteria);
+
+	return {
+		childGroupIds,
+		criterion: criteria,
+		groupId: parentGroupId,
+		index
+	};
+}
+
+/**
+ * A function that decorates the passed in component with the drag source HOC.
+ * This was separated out since this function needed to be called again for the
+ * nested groups.
+ * @param {React.Component} component The component to decorate.
+ */
+const withDragSource = dragSource(
+	DragTypes.CRITERIA_GROUP,
+	{
+		beginDrag
+	},
+	(connect, monitor) => ({
+		connectDragPreview: connect.dragPreview(),
+		connectDragSource: connect.dragSource(),
+		dragging: monitor.isDragging()
+	})
+);
+
+class CriteriaGroup extends Component {
+	static propTypes = {
+		connectDragPreview: PropTypes.func,
+		connectDragSource: PropTypes.func,
+		criteria: PropTypes.object,
+		dragging: PropTypes.bool,
+		editing: PropTypes.bool,
+		groupId: PropTypes.string,
+		index: PropTypes.number,
+		modelLabel: PropTypes.string,
+		onChange: PropTypes.func,
+		onMove: PropTypes.func,
+		parentGroupId: PropTypes.string,
+		root: PropTypes.bool,
+		supportedConjunctions: PropTypes.array,
+		supportedOperators: PropTypes.array,
+		supportedProperties: PropTypes.array,
+		supportedPropertyTypes: PropTypes.object
+	};
+
+	static defaultProps = {
+		root: false
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.NestedCriteriaGroupWithDrag = withDragSource(CriteriaGroup);
+	}
+
+	_getConjunctionLabel(conjunctionName, conjunctions) {
+		const conjunction = conjunctions.find(
+			({name}) => name === conjunctionName
+		);
+
+		return conjunction ? conjunction.label : undefined;
+	}
+
+	_handleConjunctionClick = event => {
+		event.preventDefault();
+
+		const {criteria, onChange, supportedConjunctions} = this.props;
+
+		const index = supportedConjunctions.findIndex(
+			item => item.name === criteria.conjunctionName
+		);
+
+		const conjunctionSelected = index === supportedConjunctions.length - 1 ?
+			supportedConjunctions[0].name :
+			supportedConjunctions[index + 1].name;
+
+		onChange(
+			{
+				...criteria,
+				...{
```

Comment:
See above comment about removing the spread and braces for a single item in an object